src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -246,5 +246,50 @@ private static bool IsAssembly(string filePath)
             }
         }
     }
+
+    internal class ExcludedFilesHelper
+    {
+        const string RELATIVE_KEY = nameof(RELATIVE_KEY);
+        Dictionary<string, Matcher> _matcherDict;
+        public ExcludedFilesHelper(string[] excludes)

```

Comment:
I'm not familiar with most of the filesystem apis used in this class. Will play around with them so I can better review this portion